### DISCUSSION TITLE
Why is `ClassProp` in `VariantProps`?

### DISCUSSION THREAD

**Discussion by hasparus** [joe-bell/cva#36] [Q&A] [answered]
Howdy 👋

I'm using cva with React, and I'm declaring my props like this.

```tsx
const buttonStyle = cva("...", {
  variants: {
    intent: { primary: "..." }
  }
});

export interface ButtonProps
  extends React.ComponentPropsWithoutRef<"button">,
    Omit<VariantProps<typeof buttonStyle>, "class"> {}
    
export function Button({ className, intent, ...rest }: ButtonProps) {
  return <button className={buttonStyle({ class: className, intent })} {...rest} />;
}

```

**VariantProps** includes **ClassProp**  _`{ class?: ClassValue }`_.

I already have `className`. I'd rather avoid having a `class` prop too, so I'm using **Omit** to remove it.

It's not a big issue, but I wanted to ask if you considered a version of **VariantProps** type without the `class`.

Thank you and KUTGW :)

**Comment by joe-bell** [comment, marked as answer]
Good question! I'd be up for removing it 🕺

**Reply by joe-bell** [reply]
Forgot to mention this was fixed in [0.2.3](https://github.com/joe-bell/cva/releases/tag/v0.2.3)
